3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/adapters/registry/RemoteAdapterSpec.scala

```diff
@@ -0,0 +1,231 @@
+package com.snowplowanalytics.snowplow.enrich.common
+package adapters
+package registry
+
+import java.io.InputStream
+import java.net.InetSocketAddress
+import java.util.concurrent.TimeUnit
+
+import com.snowplowanalytics.snowplow.enrich.common.loaders.{
+  CollectorApi,
+  CollectorContext,
+  CollectorPayload,
+  CollectorSource
+}
+import com.sun.net.httpserver.{HttpExchange, HttpHandler, HttpServer}
+import org.joda.time.DateTime
+import org.json4s.DefaultFormats
+import org.json4s.JsonAST.JArray
+import org.json4s.jackson.JsonMethods.parse
+import org.json4s.JsonDSL._
+import org.json4s.jackson.Serialization.write
+import org.specs2.Specification
+import org.specs2.scalaz.ValidationMatchers
+import org.specs2.specification.{Before, BeforeAfter}
+import scalaz.Scalaz._
+import scalaz.NonEmptyList
+
+import scala.concurrent.duration.Duration
+import scala.util.control.NonFatal
+
+class RemoteAdapterSpec extends Specification with ValidationMatchers {
+
+  // Connections to external webservices are outside the scope of normal unit tests, so they are normally disabled here.
+  // But if you happen to have a test remote enrichment webservice running somewhere that behaves like the localHttpServer below,
+  // you can specify its url here:
+  private val externalEnrichmentUrl  = None //e.g. Some("http://127.0.0.1:8995/myEnrichment")
+  private val externalActionTimeout  = Duration(5, TimeUnit.SECONDS)
+  private def shouldRunExternalTests = externalEnrichmentUrl.isDefined
+
+  override def is = sequential ^ s2"""
+   This is a specification to test the RemoteAdapter functionality.
+   RemoteAdapter must return any events parsed by this local test enricher                 ${testWrapperLocal(e1)}
```

Comment:
What do you mean by `this local test enricher`?

Reply:
It was adapter. mistyped.